src/shared/components/barGraph/BarGraph.tsx

```diff
@@ -0,0 +1,203 @@
+import * as React from "react";
+import * as _ from 'lodash';
+import {ThreeBounce} from 'better-react-spinkit';
+import { CancerStudy } from 'shared/api/generated/CBioPortalAPI';
+import * as Highcharts from 'highcharts';
+
+import HCE from 'highcharts/modules/exporting';
+HCE(Highcharts);
+
+
+export interface IBarGraphProps {
+    data:CancerStudy[];
+};
+
+interface ICancerTypeStudy {
+    caseCount:number;
+    shortName:string;
+    studies: CancerStudy[];
+};
+
+interface ICounterObj {
+    [name:string]: ICancerTypeStudy;
+};
+
+export default class BarGraph extends React.Component<IBarGraphProps, {colors: string[]}> {
+
+    constructor() {
+        super();
+    }
+
+    getShortName(shortName:string) {
+        switch(shortName) {
+            case 'Ovarian':
+                return 'Ovary';
+            case 'Cervical':
+                return 'Cervix';
+            case 'Colorectal':
+                return 'Colon';
+            case 'Uterine':
+                return 'Uterus';
+            case 'Melanoma':
+                return 'Skin';
+            case 'CCLE':
+                return 'Mixed';
+            case 'Thymoma(TCGA)':
+                return 'Thymus';
+            case 'Uveal':
+                return 'Eye';
+            default:
+                return shortName;
+        }
+    }
+
+     condenseCancerTypes(x:ICancerTypeStudy, y:ICancerTypeStudy[], shortName:string) {
+        const studies = [..._.flattenDeep(_.map(y, 'studies')), ...x.studies]
+        return {
+            caseCount: y.reduce((a, b) => a + b.caseCount, x.caseCount),
+            shortName,
+            studies
+        };
+    }
+
+    shouldComponentUpdate() {
+        return false;
+    }
+
+    lightenDarkenColor(col:string, amt:number) {
+        const num = parseInt(col,16);
+
+        let r = (num >> 16) + amt;
+
+        if (r > 255) r = 255;
+        else if  (r < 0) r = 0;
+
+        let b = ((num >> 8) & 0x00FF) + amt;
+
+        if (b > 255) b = 255;
+        else if  (b < 0) b = 0;
+
+        let g = (num & 0x0000FF) + amt;
+
+        if (g > 255) g = 255;
+        else if (g < 0) g = 0;
+
+        return "#" + (g | (b << 8) | (r << 16)).toString(16);
+    }
+
+    componentDidMount() {
+        let cancerTypeStudiesObj = _.reduce(this.props.data, (counterObj:ICounterObj, study) => {
```

Comment:
this reduction should be extracted to a pure function (or method) for testing purposes